subsys/bluetooth/controller/hci/hci.c

```diff
@@ -3976,6 +3976,193 @@ static void le_per_adv_sync_established(struct pdu_data *pdu_data,
+		uint8_t data_len_curr = 0U;
+		uint8_t *data_curr = NULL;
+		uint8_t sec_phy_curr = 0U;
+		struct pdu_adv_hdr *h;
+		uint8_t *ptr;
+		uint8_t len;
+
+		/* The Link Layer currently returns RSSI as an absolute value */
+		rssi = -(node_rx_curr->hdr.rx_ftr.rssi);
+
+		BT_DBG("len = %u, rssi = %d", adv->len, rssi);
+
+		p = (void *)&adv->adv_ext_ind;
+		h = (void *)p->ext_hdr_adi_adv_data;
+		ptr = (uint8_t *)h + sizeof(*h);
+
+		BT_DBG("    Ext. adv mode= 0x%x, hdr len= %u", p->adv_mode,
+		       p->ext_hdr_len);
+
+
+		if (!p->ext_hdr_len) {
+			len = ptr - (uint8_t *)p;
+
+			goto no_ext_hdr;
+		}
+
+		/* No AdvA */
+		/* No TargetA */
+		/* No ADI */
+
+		/* AuxPtr */
+		if (h->aux_ptr) {
+			struct pdu_adv_aux_ptr *aux;
+			uint8_t aux_phy;
+
+			aux = (void *)ptr;
+			ptr += sizeof(*aux);
+
+			sec_phy_curr = aux->phy + 1;
+
+			aux_phy = BIT(aux->phy);
+
+			BT_DBG("    AuxPtr chan_idx = %u, ca = %u, offs_units "
+			       "= %u offs = 0x%x, phy = 0x%x", aux->chan_idx,
+			       aux->ca, aux->offs_units, aux->offs, aux_phy);
+		}
+
+		/* No SyncInfo */
+
+		/* Tx Power */
+		if (h->tx_pwr) {
+			tx_pwr = *(int8_t *)ptr;
+			ptr++;
+
+			BT_DBG("    Tx pwr= %d dB", tx_pwr);
+		}
+
+		len = ptr - (uint8_t *)p;
+		uint8_t hdr_len = len - offsetof(struct pdu_adv_com_ext_adv,
+					      ext_hdr_adi_adv_data);
+		if (hdr_len > p->ext_hdr_len) {
+			BT_WARN("    Header length %u/%u, INVALID.", hdr_len,
+				p->ext_hdr_len);
+		} else {
+			uint8_t acad_len = p->ext_hdr_len - hdr_len;
+
+			if (acad_len) {
+				ptr += acad_len;
+				len += acad_len;
+
+				BT_DBG("ACAD: <todo>");
+			}
+		}
+
+no_ext_hdr:
+		if (len < adv->len) {
+			data_len_curr = adv->len - len;
+			data_curr = ptr;
+
+			BT_DBG("    AD Data (%u): <todo>", data_len);
+		}
+
+		if (node_rx_curr == node_rx) {
+			data_len = data_len_curr;
+			total_data_len = data_len;
+			data = data_curr;
+		} else {
+			/* TODO: Validate current value with previous ??
+			 */
+
+			if (!data) {
+				data_len = data_len_curr;
+				total_data_len = data_len;
+				data = data_curr;
+			} else {
+				total_data_len += data_len_curr;
+
+				/* TODO: construct new HCI event for this
+				 * fragment.
+				 */
+			}
+		}
+
+		if (!node_rx_next) {
+			if (sec_phy_curr) {
+				data_status = BIT(1);
+			}
+
+			break;
+		}
+
+		node_rx_curr = node_rx_next;
+		node_rx_next = node_rx_curr->hdr.rx_ftr.extra;
+		adv = (void *)node_rx_curr->pdu;
+	} while (1);
+
+	/* FIXME: move most of below into above loop to dispatch fragments of
+	 * data in HCI event.
+	 */
+
+	/* If data complete */
+	if (!data_status) {
+		uint8_t data_max_len;
+
+		data_max_len = CONFIG_BT_DISCARDABLE_BUF_SIZE -
+			       BT_HCI_ACL_HDR_SIZE - sizeof(*sep);
+
+		/* if data cannot fit the event, mark it as incomplete */
+		if (data_len > data_max_len) {
+			data_len = data_max_len;
+			data_status = BIT(0);
+		}
+	} else {
+		/* Data incomplete and no more to come */
+		if ((tx_pwr == 0x7f) && !data) {
```

Comment:
Seeing the value 0x7F for RSSI and Tx Power so many places, I'm surprised that we don't have a `#define` for that either. Something we could consider.